client/web/src/enterprise/batches/create/workspaces-preview/useWorkspaces.ts

```diff
@@ -0,0 +1,63 @@
+import { dataOrThrowErrors } from '@sourcegraph/http-client'
+import {
+    useConnection,
+    UseConnectionResult,
+} from '@sourcegraph/web/src/components/FilteredConnection/hooks/useConnection'
+
+import {
+    Scalars,
+    PreviewBatchSpecWorkspaceFields,
+    BatchSpecWorkspacesPreviewResult,
+    BatchSpecWorkspacesPreviewVariables,
+} from '../../../../graphql-operations'
+import { WORKSPACES } from '../backend'
+
+export const WORKSPACES_PER_PAGE_COUNT = 100
+
+export interface WorkspacePreviewFilters {
+    search: string | null
+}
+
+/**
+ * Custom hook to query the connection of `Workspaces` resolved to preview which
+ * repositories a batch spec is targeting.
```

Comment:
I had some trouble parsing this. How about something like (possibly misleading?):

```suggestion
 * Custom hook that wraps `useConnection` to resolve the workspaces used by a
 * particular batch spec.
```

Reply:
Yeah good call, I'll clean this up. 🙂